xls/dslx/stdlib/std.x

```diff
@@ -177,6 +177,53 @@ fn iterative_div_test () {
   ()
 }
 
+// Binary Division algorithm as outlined by:
+// http://www.dragonwins.com/domains/GetTechEd/de248/binary_division.htm
+fn iterative_binary_div<N: u32, DN: u32 = {N * u32:2}>(dividend: bits[N], divisor: bits[N]) -> (bits[N], bits[N]) {
```

Comment:
@grebe commented in https://github.com/google/xls/pull/935

> Do you think it would be much more work to let dividend and divisor have different widths? I think you'd just adjust the remainder width, perhaps extending when you do the comparison. Maybe that would add some special cases when divisor width > dividend width? I haven't thought through it all, and no worries if you'd rather not worry about it for now and just want to submit what's here.

@ericastor also commented:

> Agreed that it's not too complicated; actually, we have an implementation of a full arbitrary-width integer DivMod (both signed & unsigned) as part of the AbstractEvaluator inside of XLS here: https://github.com/google/xls/blob/fa7af6cbda69d02d2d50b28fa5d95f51fe8cdfb1/xls/ir/abstract_evaluator.h#L513-L581
> It should be much nicer in DSLX!

Reply:
ported the IR arbitrary-width version you [referenced](https://github.com/google/xls/blob/fa7af6cbda69d02d2d50b28fa5d95f51fe8cdfb1/xls/ir/abstract_evaluator.h#L513-L581) to DSLX.